File: basic_library/src/main/java/com/xiaoxin/library/base/BaseBackActivity.java
package com.xiaoxin.library.base;

import android.os.Bundle;

import com.blankj.swipepanel.SwipePanel;
import com.blankj.utilcode.util.SizeUtils;
import com.xiaoxin.library.R;

/**
 * @author: xiaoxin
 * date: 2019/7/19
 * describe: 侧滑返回 activity
 * 修改内容:
 */
public abstract class BaseBackActivity extends BaseActivity {

    /**
     * 返回true 侧滑返回
     * @return
     */
    public abstract boolean isSwipeBack();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        findViewById(android.R.id.content).setBackgroundColor(getResources().getColor(R.color.mediumGray));
        initSwipeBack();
    }

    private void initSwipeBack() {
        if (isSwipeBack()) {
            final SwipePanel swipeLayout = new SwipePanel(this);
            swipeLayout.setLeftDrawable(R.drawable.base_back);
            swipeLayout.setLeftEdgeSize(SizeUtils.dp2px(100));
            swipeLayout.wrapView(findViewById(android.R.id.content));
            swipeLayout.setOnFullSwipeListener(new SwipePanel.OnFullSwipeListener() {
                @Override
                public void onFullSwipe(int direction) {
                    swipeLayout.close(direction);
                    finish();
                }
            });
        }
    }


}

File: basic_library/src/main/java/com/xiaoxin/library/base/BaseBackActivity.kt
package com.xiaoxin.library.base

import android.os.Bundle
import android.view.View

import com.blankj.swipepanel.SwipePanel
import com.blankj.utilcode.util.SizeUtils
import com.xiaoxin.library.R

/**
 * @author: xiaoxin
 * date: 2019/7/19
 * describe: 侧滑返回 activity 需要NoActionBar主题，
 * 修改内容:
 */
abstract class BaseBackActivity : BaseActivity() {

    /**
     * 返回true 侧滑返回
     * @return
     */
    abstract val isSwipeBack: Boolean

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        findViewById<View>(android.R.id.content).setBackgroundColor(resources.getColor(R.color.mediumGray))
        initSwipeBack()
    }

    private fun initSwipeBack() {
        if (isSwipeBack) {
            val swipeLayout = SwipePanel(this)
            swipeLayout.setLeftDrawable(R.drawable.base_back)
            swipeLayout.setLeftEdgeSize(SizeUtils.dp2px(100f))
            swipeLayout.wrapView(findViewById<View>(android.R.id.content))
            swipeLayout.setOnFullSwipeListener { direction ->
                swipeLayout.close(direction)
                finish()
            }
        }
    }


}
